java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/consumer/DateConsumer.java

```diff
@@ -101,7 +102,7 @@ public NonNullableDateConsumer(DateMilliVector vector, int index, Calendar calen
     public void consume(ResultSet resultSet) throws SQLException {
       Date date = calendar == null ? resultSet.getDate(columnIndexInResultSet) :
           resultSet.getDate(columnIndexInResultSet, calendar);
-      vector.setSafe(currentIndex, date.getTime());
+      vector.setSafe(currentIndex, (int) (date.getTime() / DateDayVector.MILLIS_PER_DAY));
```

Comment:
are SQL dates guaranteed to have the same range as DateDayVector? (i.e. should you check for overflow?)

Reply:
Good suggestion. I have added logic to check overflow.